# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/lara-taan/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# Week 1 — Research Question

## 1) My lane and why

### Provisional lane: CTR / Opportunity Scoring

I am provisionally choosing the CTR / opportunity-scoring lane.

The goal is to identify pages where the available search-performance data suggests
there may be an opportunity to improve organic click-through performance.

This lane connects the available page-level search data with a practical decision:
which pages should be investigated or prioritized for improvement.

I am keeping this as a provisional choice because the framing can be refined as I
learn more about the data and the business context during the next few weeks.

I chose the Refresh / Content Opportunity Scoring lane.

The goal is to help content teams decide which pages should be reviewed first for possible improvement. Instead of manually reviewing thousands of pages, this project will create a ranked list of content items using observable search and engagement signals. The output will support human decision-making by highlighting pages with potential opportunities for refresh, optimization, or monitoring.

This lane is valuable because content teams have limited time and need a data-driven way to prioritize where to focus their effort.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## 2) The question: decision, action, and cost of a wrong call

### Research question

Which pages appear to have a meaningful opportunity to improve organic
click-through performance, based on their available search-performance signals?

### Unit of analysis

The unit of analysis is a **page**.

Each row represents a page-level observation and its associated search-performance
features.

### Decision

The decision is whether a page should be prioritized for further investigation
or possible content/search-snippet improvement.

### Output

The output will eventually be an opportunity score or ranked list of pages that
appear more likely to have improvement potential.

### Action

An SEO/content team could use the output to decide which pages to review first.
For example, they could inspect the page's title, description, content alignment,
search intent, and query coverage before deciding whether to make an update.

### Cost of a wrong call

A false positive could cause the team to spend time improving a page that has
little opportunity for improvement.

A false negative could cause a genuinely useful page to be overlooked, potentially
leaving an improvement opportunity unexplored.

Therefore, the model should support prioritization rather than automatically
deciding that a page must be changed.

The decision this project improves is: which content pages should be reviewed first for potential refresh opportunities.

The person acting on the output is a content strategist or SEO specialist. They can use the ranked list to decide which pages need investigation, such as updating content, improving engagement, or monitoring performance.

A wrong recommendation can waste editing time on pages that do not need improvement, while missing pages that have real opportunities. Therefore, the system should prioritize accurate ranking and provide explanations for why a page was selected.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
import pandas as pd

df = pd.read_csv("content_refresh_anonymized.csv")

print("Number of content items:", len(df))
print("Number of clients:", df["client_id"].nunique())

print("\nTrend distribution:")
print(df["trend_direction"].value_counts())

Number of content items: 30000
Number of clients: 32

Trend distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


In [5]:
# Show columns so we can identify the relevant CTR/opportunity fields

print(df.columns.tolist())

['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


The dataset contains 30,000 anonymized content items across 32 clients.

The starter dataset includes observable search and engagement signals such as impressions, CTR, average position, sessions, and trend measurements.

A first observation is that content performance is not uniform: pages have different trend directions, meaning some pages are improving while others are declining. This supports the need for a ranking system that helps prioritize review efforts.

In [6]:
print("Number of rows:", len(df))

if "ctr_gap" in df.columns:
    print("CTR gap non-null values:", df["ctr_gap"].notna().sum())
    print("Average CTR gap:", round(df["ctr_gap"].mean(), 4))
    print("Median CTR gap:", round(df["ctr_gap"].median(), 4))

Number of rows: 30000


In [22]:
declining_rate = (df["trend_direction"] == "down").mean()

print("Percentage of declining pages:", round(declining_rate * 100, 2), "%")

Percentage of declining pages: 54.21 %


Around 54.21% of pages show a downward trend in the dataset, indicating that identifying which pages require attention is a meaningful prioritization problem.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project will provide observed and measured insights from anonymized search performance data. The output will be decision-support: it can help prioritize pages for human review based on patterns in the available signals.

However, this project cannot prove Google's ranking algorithm, cannot prove that a specific content change will cause recovery, and cannot claim that a ranked page is guaranteed to improve after a refresh.

## What the starter data tells me

The starter dataset contains page-level observations that can be inspected for
search-performance and CTR-related signals.

The numbers shown above are calculated directly from the starter dataset rather
than being assumptions. They provide an initial indication of whether there is
enough variation and coverage in the data to investigate the CTR opportunity
question further.

These figures are evidence that the lane is worth investigating, but they do not
yet prove that changing a page will improve CTR.

## 4) Careful words: what I can and can't claim

### What I can claim

- The dataset contains page-level observations that can be analyzed for
  search-performance patterns.
- The available signals can be used to identify pages that may deserve further
  investigation.
- A ranking or score could help a team prioritize its limited review time.
- The starter data gives enough initial evidence to investigate this question
  during the capstone.

### What I cannot claim yet

- I cannot claim that a high opportunity score will definitely lead to higher CTR.
- I cannot claim that the model will establish causation.
- I cannot claim that every recommended page should be changed.
- I cannot claim that the available dataset represents every website or every
  future search situation.
- I cannot claim that ML will outperform a simple rule until that comparison is
  actually tested.

The eventual model should therefore be treated as a decision-support tool rather
than an automatic decision-maker.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.

## 5) Self-check

- [x] I selected a provisional lane.
- [x] I defined the unit of analysis.
- [x] I stated the research question.
- [x] I identified the decision.
- [x] I identified the action someone could take.
- [x] I described the cost of a wrong recommendation.
- [x] I will use real numbers from the starter dataset.
- [x] I explained why the problem is more than simply training a model.
- [x] I used careful language and avoided claiming causation.
- [x] I understand that the lane can be changed later if the evidence suggests
      another direction is more useful.